util/throttle.cc

```diff
@@ -140,6 +142,13 @@ ThrottleThread(
             tot_compact+=gThrottleData[loop].m_Compactions;
         }   // for
 
+        // capture current state of level-0 and other levels' backlog
+        gThrottleData[replace_idx].m_Backlog=gCompactionThreads->m_WorkQueueAtomic;
+        gPerfCounters->Add(ePerfThrottleBacklog1, gThrottleData[replace_idx].m_Backlog);
+
+        gThrottleData[0].m_Backlog=gLevel0Threads->m_WorkQueueAtomic;
```

Comment:
gThrottleData[0] is accessed in several places outside the lock, which means we could have an inconsistent state for gThrottleData[0] if someone calls SetThrottleWriteRate() for level 0 while we're taking care of business (e.g., on line 179); before releasing the lock on line 126, should make a copy of gThrottleData[0] and clear it (moving the memset on line 214) and then access the copy outside the lock